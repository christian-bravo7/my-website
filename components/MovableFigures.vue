<template>
  <div class="absolute w-full h-full pointer-events-none">
    <figure
      class="absolute w-6 h-6 text-pink-800 dark:text-blue-700 figure figure--1"
      :class="currentRoute"
    >
      <TriangleSolid class="fill-current w-full h-full opacity-50 md:opacity-100" />
    </figure>
    <figure
      class="absolute w-6 h-6 text-pink-800 dark:text-blue-500 figure figure--2"
      :class="currentRoute"
    >
      <TriangleOutlined class="fill-current w-full h-full opacity-50 md:opacity-100" />
    </figure>
    <figure
      class="absolute w-6 h-6 text-pink-800 dark:text-blue-500 figure figure--3"
      :class="currentRoute"
    >
      <SquareOutlined class="fill-current w-full h-full opacity-50 md:opacity-100" />
    </figure>
  </div>
</template>

<script lang="ts">
import { Vue, Component, Prop } from 'vue-property-decorator';

import TriangleSolid from '~/assets/img/triangle-solid.svg?inline';
import TriangleOutlined from '~/assets/img/triangle-outlined.svg?inline';
import SquareSolid from '~/assets/img/square-solid.svg?inline';
import SquareOutlined from '~/assets/img/square-outlined.svg?inline';

@Component({
  components: {
    TriangleSolid,
    TriangleOutlined,
    SquareSolid,
    SquareOutlined,
  },
})
export default class MovableFigures extends Vue {
  @Prop({ type: String, required: true })
  currentRoute!: string;
}
</script>

<style lang="scss" scoped>
.figure {
  $self: &;

  transition: color 200ms, top 6s, left 6s, transform 6s;

  &.index {
    &#{$self}--1 {
      top: 25%;
      left: 80%;
      transform: rotate(35deg);
    }
    &#{$self}--2 {
      top: 20%;
      left: 20%;
      transform: rotate(-35deg);
    }
    &#{$self}--3 {
      top: 80%;
      left: 60%;
      transform: rotate(100deg);
    }
  }

  &.me {
    &#{$self}--1 {
      top: 25%;
      left: 50%;
      transform: rotate(0deg);
    }
    &#{$self}--2 {
      top: 80%;
      left: 30%;
      transform: rotate(-100deg);
    }
    &#{$self}--3 {
      top: 50%;
      left: 75%;
      transform: rotate(-80deg);
    }
  }

  &.blog {
    &#{$self}--1 {
      top: 35%;
      left: 10%;
      transform: rotate(0deg);
    }
    &#{$self}--2 {
      top: 90%;
      left: 50%;
      transform: rotate(-100deg);
    }
    &#{$self}--3 {
      top: 20%;
      left: 85%;
      transform: rotate(-80deg);
    }
  }
}
</style>
