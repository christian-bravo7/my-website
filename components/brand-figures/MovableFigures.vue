<template>
  <div class="hidden md:block absolute overflow-hidden w-full h-full pointer-events-none">
    <figure
      class="absolute w-48 h-48 rounded-full figure figure--1"
      :class="currentRoute"
    />
    <figure
      class="absolute w-32 h-32 rounded-full figure figure--2"
      :class="currentRoute"
    />
    <figure
      class="absolute w-20 h-20 rounded-full figure figure--3"
      :class="currentRoute"
    />
  </div>
</template>

<script lang="ts">
import { Vue, Component, Prop } from 'vue-property-decorator';

@Component
export default class MovableFigures extends Vue {
  @Prop({ type: String, required: true })
  currentRoute!: string;
}
</script>

<style lang="scss" scoped>

.figure--1 {
  background: linear-gradient(210deg, #FFFFFF1A 0%, #FE65651A 58%, #A40EB033 100%);
}

.figure--2 {
  background: linear-gradient(210deg, #FFFFFF1A 0%, #A8EDE938 58%, #FED6E385 100%);
}

.figure--3 {
  background: linear-gradient(210deg, #FFFFFF1A 0%, #C7EAFD3F 58%, #E8198B28 100%);
}

.figure {
  $self: &;

  transition: color 200ms, top 6s, left 6s, transform 6s;

  &.index {
    &#{$self}--1 {
      top: 30%;
      left: 92%;
    }
    &#{$self}--2 {
      top: 20%;
      left: 10%;
    }
    &#{$self}--3 {
      top: 80%;
      left: 60%;
    }
  }

  &.me {
    &#{$self}--1 {
      top: 20%;
      left: 80%;
    }
    &#{$self}--2 {
      top: 50%;
      left: 15%;
    }
    &#{$self}--3 {
      top: 90%;
      left: 70%;
    }
  }

  &.blog {
    &#{$self}--1 {
      top: 10%;
      left: 88%;
    }
    &#{$self}--2 {
      top: 60%;
      left: 5%;
    }
    &#{$self}--3 {
      top: 94%;
      left: 50%;
    }
  }
}

html.dark-mode {
  .figure {
    &::after {
      position: absolute;
      top: 50%;
      right: 0;
      bottom: 0;
      left: 50%;
      width: 97%;
      height: 97%;
      background: theme('colors.blue.800');
      border-radius: 50%;
      transform: translate(-50%, -50%);
      content: '';
    }
  }

  .figure--1 {
    background: linear-gradient(210deg, rgba(255, 255, 255, 0.46) 0%, rgba(254, 101, 101, 0.52) 58%, rgba(164, 14, 176, 0.86) 100%);
  }

  .figure--2 {
    background: linear-gradient(210deg, rgba(255, 255, 255, 0.62) 0%, rgba(168, 237, 233, 0.27) 58%, rgba(255, 71, 131, 0.6) 100%);
  }

  .figure--3 {
    background: linear-gradient(210deg, rgba(255, 255, 255, 0.44) 0%, rgba(199, 234, 253, 0.17) 58%, rgba(232, 25, 139, 0.86) 100%);
  }
}
</style>
