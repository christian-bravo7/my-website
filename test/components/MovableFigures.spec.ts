import { mount } from '@vue/test-utils';

import MovableFigures from '@/components/MovableFigures.vue';

describe('MovableFigures component', () => {
  it('should render correctly', () => {
    const wrapper = mount(MovableFigures, {
      propsData: {
        currentRoute: 'about',
      },
    });

    expect(wrapper.html()).toMatchSnapshot();
  });

  it('should set class name equals to currentRoute', () => {
    const wrapper = mount(MovableFigures, {
      propsData: {
        currentRoute: 'home',
      },
    });

    const figures = wrapper.findAll('figure.absolute');
    const figuresWithClass = figures.filter(wrapper => wrapper.classes('home'));

    expect(figures.length).toBe(figuresWithClass.length);
  });
});
