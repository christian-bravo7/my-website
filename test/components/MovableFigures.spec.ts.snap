// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovableFigures component should render correctly 1`] = `
<div class="absolute w-full h-full pointer-events-none">
  <figure class="absolute w-6 h-6 text-pink-800 dark:text-blue-700 figure figure--1 about"><svg class="fill-current w-full h-full opacity-50 md:opacity-100"></svg></figure>
  <figure class="absolute w-6 h-6 text-pink-800 dark:text-blue-500 figure figure--2 about"><svg class="fill-current w-full h-full opacity-50 md:opacity-100"></svg></figure>
  <figure class="absolute w-6 h-6 text-pink-800 dark:text-blue-500 figure figure--3 about"><svg class="fill-current w-full h-full opacity-50 md:opacity-100"></svg></figure>
</div>
`;
